docs/docsite/rst/user_guide/playbooks_handlers.rst

```diff
@@ -70,10 +74,47 @@ In this example playbook, the second task notifies the handler. A single task ca
+
+
+Naming handlers
+---------------
+
+Handlers must be named in order for tasks to be able to notify them using the ``notify`` keyword.
+
+Alternatively handlers can utilize the ``listen`` keyword. Using this handler keyword handlers can listen on topics that can group multiple handlers as follows:
+
+.. code-block:: yaml
+
+    tasks:
+      - name: Restart everything
+        command: echo "this task will restart the web services"
+        notify: "restart web services"
+    handlers:
+      - name: Restart memcached
+        service:
+          name: memcached
+          state: restarted
+        listen: "restart web services"
+      - name: Restart apache
+        service:
+          name: apache
+          state: restarted
+        listen: "restart web services"
+
+Notifying ``restart web services`` topic will result in executing all handlers listening to that topic regardless of how those handlers are named.
+
+This use makes it much easier to trigger multiple handlers. It also decouples handlers from their names, making it easier to share handlers among playbooks and roles (especially when using 3rd party roles from a shared source like Galaxy).
+
+Each handler should have a globally unique name. If multiple handlers with the same name are defined only the last one defined can be notified via ``notify``, effectively shadowing all of the previous handlers with the same name. Alternatively handlers sharing the same name can all be notified and executed if they listen on the same topic by notifying that topic.
+
+There is only one global scope for handlers (handler names and listen topics) regardless of where the handlers are defined. This includes handlers defined in roles as well.
+
+
 Controlling when handlers run
 -----------------------------
 
-By default, handlers run after all the tasks in a particular play have been completed. This approach is efficient, because the handler only runs once, regardless of how many tasks notify it. For example, if multiple tasks update a configuration file and notify a handler to restart Apache, Ansible only bounces Apache once to avoid unnecessary restarts.
+By default, handlers run after all the tasks in a particular play have been completed. Notified handlers are executed automatically after each of the following sections: ``pre_tasks``, ``tasks`` and ``post_tasks``. This approach is efficient, because the handler only runs once, regardless of how many tasks notify it. For example, if multiple tasks update a configuration file and notify a handler to restart Apache, Ansible only bounces Apache once to avoid unnecessary restarts.
```

Comment:
also phrase that the 'multiple tasks' are 'in a section', handler A can be executed 4 times, if it gets notified in each section

Reply:
> also phrase that the 'multiple tasks' are 'in a section', handler A can be executed 4 times, if it gets notified in each section

It would be just 3 times because `roles` and `tasks` share the same implicit `flush_handers` task. Of course it might be more when `flush_handlers` is used within each of those sections, even multiple times.

I have that mentioned in the last sentence of `Controlling when handlers run` though, should that sentence be clarified more?